### ISSUE TITLE
[T-5.5] Integrate kernel into worker pipeline

### ISSUE THREAD

**Issue by Kobe16** [Kobe16/crucible#58]
## Task: Integrate kernel into worker pipeline

**Sprint:** 5 — CUDA/Triton Kernels
**Epic:** E5 — Custom CUDA/Triton Kernels
**Est:** 1 hour

### Subtasks

- [ ] Add config flag: `USE_CUSTOM_KERNEL=true/false`
- [ ] In `model_runner.py`, swap softmax implementation:
  ```python
  if config.use_custom_kernel and torch.cuda.is_available():
      from cuda_kernels import fused_softmax
  else:
      fused_softmax = torch.nn.functional.softmax
  ```
- [ ] Verify worker runs correctly with both settings
